app/models/katello/gpg_key.rb

```diff
@@ -6,21 +6,38 @@ class GpgKey < Katello::Model
 
     has_many :repositories, :class_name => "Katello::Repository", :inverse_of => :gpg_key, :dependent => :nullify
     has_many :products, :class_name => "Katello::Product", :inverse_of => :gpg_key, :dependent => :nullify
-
+    has_many :ssl_ca_products, :class_name => "Katello::Product", :foreign_key => "ssl_ca_cert_id",
+                               :inverse_of => :ssl_ca_cert, :dependent => :nullify
+    has_many :ssl_client_products, :class_name => "Katello::Product", :foreign_key => "ssl_client_cert_id",
+                                   :inverse_of => :ssl_client_cert, :dependent => :nullify
+    has_many :ssl_key_products, :class_name => "Katello::Product", :foreign_key => "ssl_client_key_id",
+                                :inverse_of => :ssl_client_key, :dependent => :nullify
+    has_many :ssl_ca_repos, :class_name => "Katello::Repository", :foreign_key => "ssl_ca_cert_id",
+                            :inverse_of => :ssl_ca_cert, :dependent => :nullify
+    has_many :ssl_client_repos, :class_name => "Katello::Repository", :foreign_key => "ssl_client_cert_id",
+                                :inverse_of => :ssl_client_cert, :dependent => :nullify
+    has_many :ssl_key_repos, :class_name => "Katello::Repository", :foreign_key => "ssl_client_key_id",
+                             :inverse_of => :ssl_client_key, :dependent => :nullify
     belongs_to :organization, :inverse_of => :gpg_keys
 
     validates_lengths_from_database
     validates :name, :presence => true, :uniqueness => {:scope => :organization_id,
                                                         :message => N_("has already been taken")}
+    validates :content_type, :presence => true, :inclusion => { :in => %w(gpg_key cert),
```

Comment:
Could you make these two types Constants in this file?  That way we can refer to the constants instead of the hard coded strings?  Similar to repo types in repository.rb